components/brave_wallet_ui/nft/components/nft-content/nft-content.tsx

```diff
@@ -25,29 +25,45 @@ interface Props {
 }
 
 export const NftContent = (props: Props) => {
-  const {
-    nftMetadata,
-    imageUrl,
-    displayMode
-  } = props
+  const { nftMetadata, imageUrl, displayMode } = props
+  const wrapperRef = React.useRef<HTMLDivElement>(null)
+  const [isInView, setIsInView] = React.useState<boolean>(false)
 
   const url = React.useMemo(() => {
     return stripChromeImageURL(imageUrl) ?? Placeholder
   }, [imageUrl])
 
+  const onIntersection = React.useCallback(
+    (entries: IntersectionObserverEntry[]) => {
+      const [entry] = entries
+      if (entry.isIntersecting) {
+        setIsInView(true)
+      }
+    },
+    []
+  )
+
+  React.useEffect(() => {
+    if (!wrapperRef?.current) return
+    const observer = new IntersectionObserver(onIntersection, {})
+    observer.observe(wrapperRef?.current)
```

Comment:
Shouldn't need `?` since you exited early if not truthy

```suggestion
    observer.observe(wrapperRef.current)
```